src/utils.jl

```diff
@@ -0,0 +1,32 @@
+"""
+    safe_div(x, y)
+
+Safely divide `x` by `y`. If `y` is zero, return `x` directly.
+"""
+safe_div(x, y) = ifelse(iszero(y), x, x/y)
+
+"""
+    least_dims(idxs)
+
+Compute the least dimensions, of which array can be accessed by the indices `idxs`.
+"""
+least_dims(idxs::AbstractArray{<:Integer}) = (maximum(idxs), )
+
+function least_dims(idxs::AbstractArray{<:Tuple})
+    Tuple(maximum(xs) for xs in zip(idxs...))
+end
+
+_check_input(idx::AbstractArray{<:Integer}, arr) = checkbounds(arr, minimum(idx):maximum(idx))
+
+function _check_input(idx::AbstractArray{Tuple}, arr)
+    pairs = map(xs -> Base.OneTo(maximum(xs)), zip(idx...))
+    checkbounds(arr, pairs...)
+end
```

Comment:
Does this work?

I think you want something like this, I haven't tried but believe it ought to work on the GPU, someone should check though:
```
idx = [(1,2), (3,4), (2,100), (-100,3)]

lows, highs = reduce(idx; init=(first(idx), first(idx))) do (lows, highs), xs
    map(min, xs, lows), map(max, xs, highs)
end

checkbounds(arr, CartesianIndex(lows))
checkbounds(arr, CartesianIndex(highs))
```
Also, with `dims` this won't be right, as the check should be only against some of the axes of the array. So perhaps the function should be more like `_check_bounds(axes(arr)[dims:end], idx)` --- pass the axes not the whole array.

Reply:
Could you elaborate this clearly? Maybe I just cannot get your point.

If the point is questioning about the functionality of `_check_input` function, we do have test on this.
It should work well or throw exceptions such that the test case should catch exceptions correctly.

If the point is the implementation of checking, not passing the whole array but their axes.
I still not understand the reason passing the whole array would be expensive. Maybe there are some other considerations.